File: Data Structures & Algorithms/valid-palindrome-ii/submission-0.cpp
class Solution {
    bool isPalindromeHelper(string& s, int i, int j){
        while(i <= j){
            if(s[i]!=s[j]){
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
public:
    bool validPalindrome(string s) {
        int left = 0, right = s.length() -1;
        while(left < right){
            if(s[left] != s[right]){
                return isPalindromeHelper(s, left + 1, right) || isPalindromeHelper(s, left, right - 1);
            }
            left++;
            right--;
        }
        return true;
    }
};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def __init__(self):
        self.count = 0
    
    def traverse(self, root: TreeNode, maxValInPath: int) -> None:
        if not root:
            return
        if root.val >= maxValInPath:
            self.count+=1
        maxValInPath = max(maxValInPath, root.val)
        self.traverse(root.left, maxValInPath)
        self.traverse(root.right, maxValInPath)
        
    def goodNodes(self, root: TreeNode) -> int:
        self.count = 0
        maxValInPath = -101
        self.traverse(root, maxValInPath)
        return self.count
